client/state/ui/editor/contact-form/reducer.js

```diff
@@ -2,37 +2,61 @@
  * External dependencies
  */
 import cloneDeep from 'lodash/cloneDeep';
+import merge from 'lodash/merge';
 
 /**
  * Internal dependencies
  */
+import i18n from 'lib/mixins/i18n';
 import {
-	EDITOR_CONTACT_FORM_LOAD_FORM,
-	EDITOR_CONTACT_FORM_ADD_DEFAULT_FIELD,
-	EDITOR_CONTACT_FORM_REMOVE_FIELD,
-	EDITOR_CONTACT_FORM_CLEAR_FORM
+	EDITOR_CONTACT_FORM_CLEAR,
+	EDITOR_CONTACT_FORM_LOAD,
+	EDITOR_CONTACT_FORM_FIELD_ADD,
+	EDITOR_CONTACT_FORM_FIELD_REMOVE,
+	EDITOR_CONTACT_FORM_FIELD_UPDATE,
+	EDITOR_CONTACT_FORM_SETTINGS_UPDATE
 } from 'state/action-types';
 import { CONTACT_FORM_DEFAULT, CONTACT_FORM_DEFAULT_NEW_FIELD } from './constants';
 
-const initalState = cloneDeep( CONTACT_FORM_DEFAULT );
+const initialState = cloneDeep( CONTACT_FORM_DEFAULT );
+
+export default function( state = initialState, action ) {
+	const { index, field, settings } = action;
 
-export default function( state = initalState, action ) {
 	switch ( action.type ) {
```

Comment:
Please, also add in cases for `SERIALIZE` and `DESERIALIZE`. Without them this subtree will persist by default. We can skip persistence by returning an empty object for each:

```javascript
case SERIALIZE:
    return {};
case DESERIALIZE:
   return {};
```

(Also see: https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#data-persistence--2754-)